spinnaker/spinnaker_system/kube_find_image_test.py

```diff
@@ -0,0 +1,519 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration test to see if the image promotion process is working for the
+Spinnaker Kubernetes integration.
+
+See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+for more details.
+
+The test will use ssh to peek at the spinnaker configuration
+to determine the managed project it should verify, and to determine
+the spinnaker account name to use when sending it commands.
+
+Sample Usage:
+    Assuming you have created $PASSPHRASE_FILE (which you should chmod 400)
+    and $CITEST_ROOT points to the root directory of this repository
+    (which is . if you execute this from the root)
+
+  PYTHONPATH=$CITEST_ROOT:$CITEST_ROOT/spinnaker \
+    python $CITEST_ROOT/spinnaker/spinnaker_system/kube_find_image_test.py \
+    --gce_ssh_passphrase_file=$PASSPHRASE_FILE \
+    --gce_project=$PROJECT \
+    --gce_zone=$ZONE \
+    --gce_instance=$INSTANCE
+or
+  PYTHONPATH=$CITEST_ROOT:$CITEST_ROOT/spinnaker \
+    python $CITEST_ROOT/spinnaker/spinnaker_system/kube_find_image_test.py \
+    --native_hostname=host-running-find-image-test
+"""
+
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.kube_testing as kube
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+
+class KubeFindImageTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the find image test.
+
+  This scenario defines the different test operations.
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Load Balancer
+    Create a Spinnaker Server Group
+    Create a Pipeline with the following stages
+      - Find Image
+      - Deploy
+      - Disable Server Group
+    Delete each of the above (in reverse order)
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 70
+    return agent
+
+  @classmethod
```

Comment:
Ah, I thought I needed this to make sure I get the regular arguments. Nothing is parsed from the config at this point since all settings are left at their default value. I could change that in the future, but that would make the config in spinnaker/spinnaker/install/first_google_boot.sh a bit more complicated than it needs to be.